### PR TITLE
pass ecc curve as a parameter to ecc functions

### PR DESCRIPTION
**Pull request by douglasbakkum**

### CODE REVIEW

File: src/ecc.c [hunk 1]
```diff
@@ -34,362 +37,350 @@
...
+    return ecc_sign_digest(curve, private_key, hash, sig);
 }
 
 
-int ecc_sign_double(const uint8_t *privateKey, const uint8_t *msg, uint32_t msg_len,
-                    uint8_t *sig)
+int ecc_sign_double(const ecc_curve_id curve, const uint8_t *privateKey,
+                    const uint8_t *msg, uint32_t msg_len, uint8_t *sig)
 {
+    (void)curve;
```
Comment (jonasschnelli) [inline src/ecc.c]: Not necessary AFAIK

File: src/ecc.c [hunk 2]
```diff
@@ -34,362 +37,350 @@
...
+void ecc_get_public_key33(const ecc_curve_id curve, const uint8_t *private_key,
+                          uint8_t *public_key)
 {
-    return uECC_verify(public_key, hash, SHA256_DIGEST_LENGTH, sig, ecc_curve);
-}
+    if (curve == ECC_CURVE_libsecp256k1) {
+        ecc_libsecp256k1_get_pubkey(private_key, public_key, 33, SECP256K1_EC_COMPRESSED);
+    }
 
+    if (curve == ECC_CURVE_uECC_nist_p) {
```
Comment (jonasschnelli) [inline src/ecc.c]: use `else if`?

File: src/ecc.c (hunk 1 above)
Comment (douglasbakkum) [reply src/ecc.c]: ack

File: src/ecc.c (hunk 2 above)
Comment (douglasbakkum) [reply src/ecc.c]: Ack. 
Here and for similar sections.

**Comment by douglasbakkum** [comment]
Superseded by #3. Closing.

**Pull request closed without merging**